Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm

```diff
@@ -267,6 +267,7 @@ + (BOOL)supportsSecureCoding
 - (instancetype)initWithCoder:(NSCoder *)aDecoder
 {
     String rawJSON = [aDecoder decodeObjectOfClass:[NSString class] forKey:@"raw_json"];
+    NSInteger dir = [aDecoder decodeIntegerForKey:@"dir"];
```

Comment:
I think for security reasons we would want this to check for integers that are out of range, and fail to decode in that case instead of making an illegal enumeration value.

Reply:
The manifest parser should guard for this , so would it be ok to just?:

```suggestion
    NSInteger dir = [aDecoder decodeIntegerForKey:@"dir"];
    ASSERT(dir >= 0 && dir <= 2);
```

Reply:
Statically, that one can only get here by having gone through the manifest parser machinery and associated call sites (coming in from `HTMLLinkElement` being responsible for loading the manifest and having processed it, `Document` having provided it... and so on). 

I get your point that there's no guarantees computationally that how we got here was from the route above (something else could have IPC'ed its way here), but that then starts applying equally to just about anything that sends data over IPC to anything in the UI Process. There's a presupposition of trust between the UI Process and WebCore. 

The presumption for the entire _WKApplicationManifest.mm file is that the data was processed through the manifest parser, and that the manifest parser conform the W3C spec, which is a strict validating parser/processor. If that invariant was to fall apart (i.e., we don't trust our own code), we would need recheck everything.

Reply:
Spoke with @geoffreygaren. Will check if we need to sprinkle MESSAGE_CHECK a bit closer to the IPC layer for the entire parsed manifest structure. 

I've filed a bug to do that as followup so we can do that all at once: 
https://bugs.webkit.org/show_bug.cgi?id=278619

Reply:
@marcoscaceres I have included your suggestion in my last push.